cli/config/parser.go

```diff
@@ -1,37 +1,57 @@
 package config
 
 import (
+	"errors"
 	"fmt"
 	"github.com/greenplum-db/gp-common-go-libs/gplog"
 	"os"
 	"strconv"
 	"strings"
 )
 
-// ReadFileProvider allows us to inject fileContents in our unit test
-var ReadFileProvider = readFile
+// ClusterFileProvider allows us to inject cluster fileContents in our unit test.
+var ClusterFileProvider = readFile
 
+// HostFileProvider allows us to inject host fileContents in our unit test.
+var HostFileProvider = readFile
+
+// ParsePxfCluster creates a PxfCluster from the given name.
+// It assumes that each clusters' configuration files are stored within $PXF_BASE/clusters/[clusterName]/conf
+// The hosts file is optional.
+// Currently, it only supports a single PxfCluster with the name "default".
 func ParsePxfCluster(clusterName string) (*PxfCluster, error) {
 	pxfDeployment := PxfDeployment{Clusters: make(map[string]*PxfCluster)}
 	clusterNames := []string{"default"}
-	path := GetPathForClusterFile(clusterNames[0])
-	fileContents, err := ReadFileProvider(path)
-	if err != nil {
-		return nil, NewPxfError(
-			// TODO: in the future add cluster name to the hint message 'pxf cluster prepare <cluster-name>'
-			"please check your PXF_BASE variable. Did you run 'pxf cluster prepare' command?",
-			"could not read configuration file %s : %w", path, err)
-	}
-	contents, err := parseFile(fileContents)
-	if err != nil {
-		return nil, NewClusterConfigError(clusterName,"failed to parse cluster configuration file : %w", err)
-	}
 	var pxfCluster *PxfCluster
 	for _, currentClusterName := range clusterNames {
-		pxfCluster, err = parseCluster(currentClusterName, contents)
+		clusterPath := GetPathForClusterFile(currentClusterName)
+		clusterFileContents, err := ClusterFileProvider(clusterPath)
+		if err != nil {
+			return nil, NewPxfError(
+				// TODO: in the future add cluster name to the hint message 'pxf cluster prepare <cluster-name>'
+				"please check your PXF_BASE variable. Did you run 'pxf cluster prepare' command?",
+				"could not read configuration file %s : %w", clusterPath, err)
+		}
+		clusterContents, err := parseClusterFileContents(clusterFileContents)
+		if err != nil {
+			return nil, NewClusterConfigError(clusterName, "failed to parse cluster configuration file : %w", err)
```

Comment:
```suggestion
			return nil, NewClusterConfigError(currentClusterName, "failed to parse cluster configuration file : %w", err)
```
We are using `currentClusterName` for each iteration